lpcompiler.pas

```diff
@@ -2605,10 +2649,8 @@ function TLapeCompiler.ParseExpression(ReturnOn: EParserTokenSet = []; FirstNext
       Result := Node;
       HasChanged := False;
 
-      if TLapeTree_Base.isEmpty(Node) or
-        (([lcoAutoInvoke, lcoAutoProperties] * Node.CompilerOptions) = []) or
-        (not (Node is TLapeTree_ExprBase)) or (Node is TLapeTree_Invoke)
-      then
+      if TLapeTree_Base.isEmpty(Node) or (not (Node is TLapeTree_ExprBase)) or (Node is TLapeTree_Invoke) or
+         ((not(lcoAutoInvoke in Node.CompilerOptions)) and (not IsProperty(TLapeTree_ExprBase(Node)))) then
```

Comment:
Might as well remove the `lcoAutoInvoke`/`IsProperty` check here and go in anyway.